test/correctness/vectorize_nested.cpp

```diff
@@ -192,6 +192,54 @@ int vectorize_all_d() {
     return 0;
 }
 
+int vectorize_inner_of_scalarization() {
+    ImageParam in(UInt(8), 2);
+
+    Var x("x_inner"), y("y");
+
+    Func out;
+    out(x, y) = in(x, y);
+
+    Var xo("xo"), yo("yo");
+    out.split(x, xo, x, 8, TailStrategy::RoundUp)
+        .split(y, yo, y, 8, TailStrategy::GuardWithIf)
+        .vectorize(x)
+        .vectorize(y);
+
+    // We are looking for a specific loop, which shouldn't have been scalarized.
+    class CheckForScalarizedLoop : public Internal::IRMutator {
+        using IRMutator::visit;
+
+        Internal::Stmt visit(const Internal::For *op) override {
+            if (Internal::ends_with(op->name, ".x_inner")) {
```

Comment:
Maybe we should also look for the y loop, so we know (a little better at least) the test isn't just not finding a loop even though it's there.

Reply:
I've added a check for y loop as well. Also, I've verified that this test fails prior to this change.